paasta_tools/paasta_maintenance.py

```diff
@@ -0,0 +1,367 @@
+    client_fn = get_schedule_client()
+    print client_fn(method="POST", endpoint="", data=json.dumps(payload)).text
+
+
+def undrain(hostnames, start, duration):
+    """Unschedules the maintenance window for the specified hosts and unmarks them as draining. They are ready for
+    regular use.
+    :param hostnames: a list of hostnames
+    :param start: the time to start the maintenance, represented as number of nanoseconds since the epoch
+    :param duration: length of the maintenance window, represented as number of nanoseconds since the epoch
+    """
+    payload = build_maintenance_schedule_payload(hostnames, start, duration, drain=False)
+    client_fn = get_schedule_client()
+    print client_fn(method="POST", endpoint="", data=json.dumps(payload)).text
+
+
+def down(hostnames):
+    """Marks the specified hostnames as being down for maintenance, and makes them unavailable for use.
+    :param hostnames: a list of hostnames
+    """
+    payload = build_start_maintenance_payload(hostnames)
+    client_fn = master_api()
+    print client_fn(method="POST", endpoint="/machine/down", data=json.dumps(payload)).text
+
+
+def up(hostnames):
+    """Marks the specified hostnames as no longer being down for maintenance, and makes them available for use.
+    :param hostnames: a list of hostnames
+    """
+    payload = build_start_maintenance_payload(hostnames)
+    client_fn = master_api()
+    print client_fn(method="POST", endpoint="/machine/up", data=json.dumps(payload)).text
+
+
+def status():
+    """Get the Mesos maintenance status. This contains hostname/ip mappings for hosts that are either marked as being
+    down for maintenance or draining.
+    """
+    status = get_maintenance_status()
+    print "%s:%s" % (status, status.text)
+
+
+def schedule():
+    """Get the Mesos maintenance schedule. This contains hostname/ip mappings and their maintenance window."""
+    schedule = get_maintenance_schedule()
+    print "%s:%s" % (schedule, schedule.text)
+
+
+def paasta_maintenance():
+    """Manipulate the maintenance state of a PaaSTA host."""
+    args = parse_args()
+
+    action = args.action
+    hostnames = args.hostname
+
+    if action not in ['drain', 'undrain', 'down', 'up', 'status']:
+        print "action must be 'drain', 'undrain', 'down', 'up', or 'status'"
+        return
+
+    if action != 'status' and not hostnames:
+        print "You must specify one or more hostnames"
+        return
+
+    start = args.start.strftime("%s")
+    duration = args.duration
+
+    if action == 'drain':
+        drain(hostnames, start, duration)
+    elif action == 'undrain':
+        undrain(hostnames, start, duration)
+    elif action == 'down':
+        down(hostnames)
+    elif action == 'up':
+        up(hostnames)
+    elif action == 'status':
+        status()
+        schedule()
+    else:
+        raise ValueError("Unknown Action: %s" % action)
```

Comment:
Correct. I added the check at 342 to fail a bit earlier. after this else statement was already here. I'll remove it.